oap-ape/ape-native/src/utils/AggExpression.h

```diff
@@ -116,7 +116,7 @@ class AggExpression : public WithResultExpression {
       }
       done = true;
     }
-    result = resultCache;
+    appendPartialResult(result, resultCache);
```

Comment:
`getResultInternalWithGroup` will need to allocate a larger vector for each batch.
I think `result` could be reused in `getResultInternalWithGroup`  to reduce memory allocations.
In that way, `appendPartialResult`'s job can be done by `getResultInternalWithGroup` or `getResultInternal`.

Reply:
Addressed this issue, and fix all null case. Please take a review, thanks.